### ISSUE TITLE
GRPO Trainer loss error? as the info log, why 'rewards/if_rewrite_reward/std': 5.080004692077637 (is not 0), but loss = 0.0, since if std is not 0, the advantages won't be 0.0, get a loss is not 0。

### ISSUE THREAD

**Issue by harmonytan** [huggingface/trl#3857]
### Reproduction

```python
from trl import GRPO
....
trainer.tain()
```

outputs:

```
```

### System Info

{'loss': 0.0, 'grad_norm': 0.0, 'learning_rate': 1.7050844078611056e-06, 'num_tokens': 1751675.0, 'completions/mean_length': 701.8125, 'completions/min_length': 307.0, 'completions/max_length': 1013.0, 'completions/clipped_ratio': 0.0, 'completions/mean_terminated_length': 701.8125, 'completions/min_terminated_length': 307.0, 'completions/max_terminated_length': 1013.0, 'rewards/if_rewrite_reward/mean': -3.0, 'rewards/if_rewrite_reward/std': 5.080004692077637, 'rewards/json_format_reward/mean': 2.0, 'rewards/json_format_reward/std': 0.0, 'rewards/think_response_format_reward/mean': 2.0, 'rewards/think_response_format_reward/std': 0.0, 'rewards/language_ratio_reward/mean': 2.0, 'rewards/language_ratio_reward/std': 0.0, 'rewards/length_penalty_reward/mean': 0.0, 'rewards/length_penalty_reward/std': 0.0, 'reward': 3.0, 'reward_std': 0.0, 'frac_reward_zero_std': 1.0, 'clip_ratio/low_mean': 0.0, 'clip_ratio/low_min': 0.0, 'clip_ratio/high_mean': 0.0, 'clip_ratio/high_max': 0.0, 'clip_ratio/region_mean': 0.0, 'epoch': 0.33}
{'loss': 0.0, 'grad_norm': 0.0, 'learning_rate': 1.6969375686552936e-06, 'clip_ratio/low_mean': 0.0, 'clip_ratio/low_min': 0.0, 'clip_ratio/high_mean': 0.0, 'clip_ratio/high_max': 0.0, 'clip_ratio/region_mean': 0.0, 'epoch': 0.33}
{'loss': 0.0, 'grad_norm': 0.0, 'learning_rate': 1.6886997750919617e-06, 'clip_ratio/low_mean': 0.0, 'clip_ratio/low_min': 0.0, 'clip_ratio/high_mean': 0.0, 'clip_ratio/high_max': 0.0, 'clip_ratio/region_mean': 0.0, 'epoch': 0.34}
{'loss': 0.0, 'grad_norm': 0.0, 'learning_rate': 1.6803721022505067e-06, 'clip_ratio/low_mean': 0.0, 'clip_ratio/low_min': 0.0, 'clip_ratio/high_mean': 0.0, 'clip_ratio/high_max': 0.0, 'clip_ratio/region_mean': 0.0, 'epoch': 0.34}

### Checklist

- [x] I have checked that my issue isn't already filed (see [open issues](https://github.com/huggingface/trl/issues?q=is%3Aissue))
- [x] I have included my system information
- [x] Any code provided is minimal, complete, and reproducible ([more on MREs](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks))
- [x] Any code provided is properly formatted in code blocks, (no screenshot, [more on code blocks](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks))
- [x] Any traceback provided is complete

**Comment by pramodith** [comment]
If `beta=0.0` and you set `num_iterations=1` the loss will always be 0 since. The current and old policy are exactly the same. This isn't a problem though GRPO loss doesn't have a lower bound i.e. the loss can be less than 0 and the gradients are non-zero even if loss is 0.0

Moreover the gradients depend on the advantage scores so the model does learn from the rewards. You can find a detailed dive into the math [here](https://github.com/huggingface/open-r1/issues/239#issuecomment-2646297851)
